### PR TITLE
Waffle Chart

### PR DESCRIPTION
**Pull request by phocks**
New chart template: Waffle chart: currently supports 10 x 10 squares. Needs to be tested with Google Drive data. Attached is the sample data that I was testing with.  
[waffle-chart.xlsx](https://github.com/abcnews/dailygraphics/files/1050891/waffle-chart.xlsx)

### CODE REVIEW

File: graphic_templates/bar_chart/js/graphic.js
Comment (drzax) [inline graphic_templates/bar_chart/js/graphic.js]: Would you mind changing the indentation in this file back to spaces?

I know the .editorconfig file had tabs specified (which was silly given the majority of existing codebase uses spaces) so that might be why this change came about. I've submitted a pull request for .editorconfig here: https://github.com/abcnews/dailygraphics/pull/72 and will merge when it gets a positive review.

File: graphic_templates/waffle_chart/css/graphic.less
Comment (drzax) [inline graphic_templates/waffle_chart/css/graphic.less]: There are some mixed tabs and spaces in this file—would be nice to keep them consistent as spaces.

File: graphic_templates/waffle_chart/graphic_config.py
Comment (drzax) [inline graphic_templates/waffle_chart/graphic_config.py]: I've created an appropriate template from this based off the pie chart template. Please update this key to 1lNeDBYiAU3EqL9Dg5YKK3qK2BBnBq1_MAgmp5pKffW0

File: graphic_templates/waffle_chart/js/graphic.js
Comment (drzax) [inline graphic_templates/waffle_chart/js/graphic.js:236]: I'd prefer to see any style declarations that aren't dynamic and based on JS variables declared in the stylesheet rather than in the javascript.

**Review by drzax — CHANGES_REQUESTED** [review]
This is looking pretty good. I have a few change requests before this gets merged back to master.

When they're done I'll also merge https://github.com/abcnews/chart-builder/pull/3 so waffle chart is available in the chart builder interface.

File: graphic_templates/waffle_chart/js/graphic.js
Comment (phocks) [reply graphic_templates/waffle_chart/js/graphic.js:236]: I agree the styles could be more css based. I saw a few online recommending "layout" based styles for D3 be controlled in the JavaScript, and "visual" like color etc. be based in the css. For the labelWidth variable as in other charts, it can be controlled via the Google Doc so if a user needs to use a large label they can set the margin so it doesn't wrap.

There's some more work to be done down the track on this.

File: graphic_templates/bar_chart/js/graphic.js
Comment (phocks) [reply graphic_templates/bar_chart/js/graphic.js]: Whoops this was just me unintentionally editing this. I've fix it. Do I need to do another pull request?

File: graphic_templates/waffle_chart/css/graphic.less
Comment (phocks) [reply graphic_templates/waffle_chart/css/graphic.less]: The original file seemed to be in tabs and my additions were the ones with spaces. I've converted all to 4 spaces.

File: graphic_templates/waffle_chart/graphic_config.py
Comment (phocks) [reply graphic_templates/waffle_chart/graphic_config.py]: Awesome thanks. This is done.

**Comment by phocks** [comment]
Okay, conflict reverted and all good.

**Review by drzax — APPROVED** [review]

**Review by blakebutcher — APPROVED** [review]

**Comment by phocks** [comment]
Oops I accidentally pushed my early Sparkline changes, but I reverted so still should be good to merge just the completed Waffle chart when ready. I will move to the abcnews version and create a new branch for sparkline-development as per GitHub Flow

**Pull request merged**
